Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/BaseDAO.java

```diff
@@ -505,4 +518,62 @@ private static List<String> getDatesFromConfig() {
                 .filter(Pattern.compile("\\d{4}|\\d{4}-\\d{2}|\\d{4}-\\d{2}-\\d{2}").asMatchPredicate())
                 .collect(Collectors.toList());
     }
+
+    /**
+     * Enters a search query into the log when it is running in debug level.
+     * Placeholders are replaced with their parameter values.
+     * 
+     * @param query
+     *            search query
+     * @param parameters
+     *            parameter values
+     */
+    private static void debugLogQuery(String query, Map<String, Object> parameters) {
+        debugLogQuery(query, parameters, Integer.MIN_VALUE, Integer.MIN_VALUE);
+    }
+
+    /**
+     * Enters a search query into the log when it is running in debug level.
+     * Placeholders are replaced with their parameter values.
+     * 
+     * @param query
+     *            search query
+     * @param parameters
+     *            parameter values
+     * @param initPointer
+     *            can initialize the object pointer to a later object (sets
+     *            {@linkplain Query#setFirstResult(int)})
+     * @param stopCount
+     *            the search stops after count hits (sets
+     *            {@linkplain Query#setMaxResults(int)})
+     */
+    private static void debugLogQuery(String query, Map<String, Object> parameters, int initPointer, int stopCount) {
+        if (logger.isDebugEnabled()) {
+            String resolved = PARAMETER_PATTERN.matcher(query).replaceAll(matchResult -> {
+                Object parameter = parameters.get(matchResult.group(1));
+                if (Objects.isNull(parameter)) {
+                    return matchResult.group();
+                }
+                if (parameter instanceof String) {
+                    return '\'' + ((String) parameter) + '\'';
+                }
+                if (parameter instanceof Collection) {
+                    int size = ((Collection<?>) parameter).size();
+                    if (size > 12) {
```

Comment:
Why is a value greater than `12` bad? Should not all parameters be used?

Reply:
I have based this on the written language: Numbers up to twelve are written out, from 13 onwards the number is written. Yes, a list of 500,000 IDs in a query makes the logging unreadable, which is why only the number of elements is output in the debug output for longer lists. I think 80% of users will be happy with this logging format, so no parameterization is required for log output.

Reply:
Thank you for the explanation. Can you add a comment with this explanation as this can lead to questions in future.